auth.php

```diff
@@ -609,27 +613,78 @@ public function saml_login() {
         return;
     }
 
+    /**
+     * Redirect SAML2 login if a flagredirecturl has been configured.
+     *
+     * @throws \moodle_exception
+     */
+    protected function redirect_blocked_access() {
+
+        if (!empty($this->config->flagredirecturl)) {
+            redirect(new moodle_url($this->config->flagredirecturl));
```

Comment:
Does `moodle_url` handle non Moodle URLS? I thought we wanted to be able to redirect users to any valid URL?

Reply:
Yeah, it could be any URL not just moodle.

Reply:
`moodle_url` doesn't have to be internal, it can be any url if it's a fully qualified domain name, or an internal url if it's relative.